packages/microcosm-dom/src/action-form.js

```diff
@@ -43,21 +43,19 @@ export function generateActionForm(createElement, Component) {
 
     submit(event) {
       let result = this.send(this.props.action, this._parameterize())
+      let action = Subject.hash(result)
```

Comment:
Wrapping the result in `Subject.hash` gives me either:

1. The same action back
2. A `Subject` version of whatever send returned. 

That means you can pass anything into the `send` prop and the action form will turn it into the same interface as an action (Subjects). You'll always get a Subject back from send now, every time.

So you don't really need a repo. Just a send method:

```javascript
let send = () => Promise.resolve(true)
let callback = (subject) => {}

<ActionButton send={send} onSend={callback}/>
```